Feature/candidate office pages

jmcarp opened the pull request:
Add candidate data views for president, senate, and house. This isn't complete, but should provide enough structure to start discussion with @noahmanger and @jenniferthibault.

Note: includes #1060.

jmcarp committed:
Update candidate totals URL.

Use URLs from https://github.com/18F/openFEC/pull/1473/.

jmcarp committed:
Add candidate data tables by office.

jmcarp committed:
Merge branch 'feature/candidate-totals-url' into feature/candidate-office-pages

noahmanger commented:
Next step: Review with Jen and me.

jmcarp committed:
Filter on cash and debt ranges.

jmcarp committed:
Add unstyled election filter.

jmcarp commented:
@noahmanger how do you want to move forward on this? It sounds like @LindsayYoung's going to review the associated patch against the API. Do you want to review https://github.com/18F/fec-style/pull/248 and this patch? Is there more I should do on these three PRs before review?

noahmanger commented on templates/partials/candidates-office-table.html:
```diff
@@ -0,0 +1,17 @@
+<table id="results" class="data-table" data-type="candidate">
```
What do you think about moving the `*-table.html` partials to just include them in the main page template for these pages? Both don't have that much code, and we never reference them anywhere else, so it could cut down on the number of files you need to create for each page.

jmcarp replied:
No strong preference, but if that change gets made, I think it can happen in a separate patch.

noahmanger replied:
:+1:

noahmanger commented on templates/candidates-office.html:
```diff
@@ -0,0 +1,34 @@
+{% block body %}
+
+<header class="page-header slab slab--primary">
+  <h1 class="page-header__title tst-page-title">Candidates</h1>
```
We're not actually using this `tst-page-title` class for testing on this page are we? Might as well remove it.

jmcarp replied:
Done.

noahmanger commented:
I'm having trouble running this locally to check it out for functionality (my local API seems to be having some data trouble). But this seems like a good place to use our new "merge unfinished work into develop and hide with feature flags if necessary" plan.

My only question I think is how this is set up to handle senate candidates. When you choose an election year and a time period, are you seeing _only_ senate candidates who filed to run that election year and then their financials for that time period? (This is what I'd expect, but just checking).

jmcarp committed:
Drop test class.

jmcarp committed:
Merge branch 'develop' of github.com:18f/openFEC-web-app into feature/candidate-office-pages

noahmanger commented on templates/partials/candidates-office-filter.html:
```diff
@@ -0,0 +1,27 @@
+{{ text.field('max_cash_on_hand_end_period', 'Maximum cash on hand', {'data-suffix': 'or less', 'data-inputmask': '"alias": "currency", "rightAlign": false, "clearMaskOnLostFocus": true'}) }}
+{{ text.field('min_debts_owed_by_committee', 'Minimum debt', {'data-suffix': 'or more', 'data-inputmask': '"alias": "currency", "rightAlign": false, "clearMaskOnLostFocus": true' }) }}
+{{ text.field('max_debts_owed_by_committee', 'Maximum debt', {'data-suffix': 'or less', 'data-inputmask': '"alias": "currency", "rightAlign": false, "clearMaskOnLostFocus": true'}) }}
+{% include 'partials/filters/hideNull.html' %}
```
Don't need hide null anymore

Commit:
Adjusting filters

- Wrapping election cycles in a fieldset with a legend
- Removing hide null filter

noahmanger commented:
Ok, I just noticed a couple small things so I included them in a PR: https://github.com/jmcarp/openFEC-web-app/pull/26

Behavior here is looking good. I had a couple notes on the fec-style patch, but this has my :+1: once my patch is merged in.

noahmanger commented on static/templates/candidates-office.hbs:
```diff
@@ -0,0 +1,35 @@
+<div class="panel__row">
+  <div class="panel__heading">
+    <h4 class="panel__title"><a href="{{basePath}}/candidate/{{candidate_id}}">{{name}}</a></h4>
```
This link should have the current cycle included in the URL.

noahmanger commented:
Spoke too soon. I'm seeing some problems that seem to be on the API side:
- Financial filters aren't working for me (seeing 422 errors)
- The name filter isn't working (no error visible) and I tried replacing the new typeahead filters but that's not working either

noahmanger commented:
Also downloads aren't working locally. Is that expected?

jmcarp committed:
Update typeahead.

jmcarp committed:
Merge branch 'election-toggle-fixes' of github.com:18f/openFEC-web-app into feature/candidate-office-pages

noahmanger commented:
Could be a separate issue (doesn't need to hold up merge), but I'll capture it here: the range filters only seem to fire after hitting enter, rather than on tabbing away, like other text input filters work.

jmcarp committed:
Fix typeahead labels.

jmcarp commented:
Try again with https://github.com/18F/openFEC/pull/1541. Downloads won't work locally unless you're running redis and a celery worker--I'm guessing nobody but @LindsayYoung or me has set up downloads to run locally.

LindsayYoung commented:
I feel like the date choices should be oldest to newest. I was super confused for a bit. 
![screen shot 2016-02-25 at 1 31 53 pm](https://cloud.githubusercontent.com/assets/4406333/13329836/2b093fdc-dbc4-11e5-9d8b-49dab15586e4.png)

jmcarp commented:
@LindsayYoung: the date choices are meant to mirror the analogous widget on the candidate detail page, which iirc is what @noahmanger and @jenniferthibault suggested. I don't have a preference about this.

LindsayYoung commented:
Another small thing is that the currency inputs will turn commas into decimals, I can live with it just wanted to flag it incase that behavior is unwanted.

jmcarp commented:
That's also the current behavior of the currency input masks throughout the site, and has been since we introduced input masking last year. Again, no preference on this--if it's a problem, I'd file a separate issue.
